.github/workflows/ci.yml

```diff
@@ -366,10 +372,13 @@ jobs:
           echo "bb-bench status: ${{ needs.bb-bench.result }}"
           echo "yarn-project-formatting status: ${{ needs.yarn-project-formatting.result }}"
           echo "yarn-project-test status: ${{ needs.yarn-project-test.result }}"
-          if [[ "${{ needs.e2e.result }}" != 'success' || "${{ needs.bb-native-tests.result }}" != 'success' || "${{ needs.bb-bench.result }}" != 'success' || "${{ needs.yarn-project-formatting.result }}" != 'success' || "${{ needs.yarn-project-test.result }}" != 'success' ]]; then
-            echo "Pull request merging not allowed due to failures."
-            exit 1
-          fi
+          for need in "${needs[@]}" 
```

Comment:
we can also get rid of the status printing above if this is working :)

Reply:
nice solution, there's probably some other clever trick here but this seems like a stable solution